### PR TITLE
feature: provide configuration for locked id response value

### PR DESCRIPTION
**Pull request by InoMurko**
Hi!
We're using ethereumex at OmiseGO and we're recording Eth client responses with ExVCR but because of the stateful protocol property (incrementing ID's) we're hacking it around with reseting counters in ETS. But I guess it could be done cleaner by locking yourself on a specific value... here's a PR that does that.

### CODE REVIEW

File: lib/ethereumex/counter.ex [hunk 1]
```diff
@@ -41,7 +41,6 @@ defmodule Ethereumex.Counter do
     :ets.update_counter(@tab, key, {2, count}, {key, 0})
   end
 
-  @spec do_increment(binary() | nil, atom(), integer()) :: integer()
```
Comment (ayrat555) [inline lib/ethereumex/counter.ex]: why did you remove this spec?

File: lib/ethereumex/counter.ex [hunk 2]
```diff
@@ -19,11 +19,29 @@ defmodule Ethereumex.Counter do
 
   @spec increment(atom()) :: integer()
   def increment(key) do
-    :ets.update_counter(@tab, key, {2, 1}, {key, 0})
+    do_increment(Application.get_env(:ethereumex, :id_lock), key)
   end
 
   @spec increment(atom(), integer()) :: integer()
   def increment(key, count) do
+    do_increment(Application.get_env(:ethereumex, :id_lock), key, count)
```
Comment (ayrat555) [inline lib/ethereumex/counter.ex]: instead of using a lock, I suggest making changes to always start the counter from zero. IIUC Because json rpc uses request ids for batch request

File: lib/ethereumex/counter.ex (hunk 1 above)
Comment (InoMurko) [reply lib/ethereumex/counter.ex]: because it's here already: https://github.com/mana-ethereum/ethereumex/pull/65/commits/35ff360b35541fbe5e7ce39236b3d6e8baa9bc02#diff-2917486c2b82f1420bd9a6a7bbaeb602L39

**Comment by ayrat555** [comment]
@InoMurko https://hex.pm/packages/ethereumex/0.5.5

**Comment by InoMurko** [comment]
Thanks @ayrat555!
